### ISSUE TITLE
SC2115 Could not expand to `/*`

### ISSUE THREAD

**Issue by stdedos** [koalaman/shellcheck#1868]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2115
- My shellcheck version (`shellcheck --version` or "online"): 
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
for d in "${DIRECT_FETCH_PATH}"* ; do
    warning 'Leftovers found in DIRECT_FETCH_PATH:'
    ls -lh "$d"
    rm -r "/$d"
done
```

#### Here's what shellcheck currently says:
```
        rm -r "/$d"
              ^-- SC2115: Use "${var:?}" to ensure this never expands to / .
```

#### Here's what I wanted or expected to see:

Nothing. I don't understand how this could expand to a "real" globbing `/*` (being quoted and not having a unquoted star nearby)

Background: while I believe that `DIRECT_FETCH_PATH="$(readlink -f "${UPLOADS_PATH}/../$(basename "$0")_wget")"` will never start with a `-` that would then be parsed by `rm` optparse, I was thinking if I should "make sure of that" or not. Prepending a '/' sounded like a reasonable idea in my head (from some other suggestion to do `./$f` for `cwd` removals).

I understand if you don't want to "try" to fix that further, because `: rm -rf /` is so bad already

**Comment by koalaman** [comment]
You're right. This can never expand to `"/"` (unless there's a weird trap or function override).

For exactly the reason you mention, SC2115 isn't clever. It just warns whenever the argument with unset variables expands to a known top level directory. Rather than trying to make it smarter and risk missing something, I'd suggest just placating ShellCheck as it suggests or ignoring the issue.
